### PR TITLE
alter default rules to include special chars

### PR DESCRIPTION
**Pull request by shakyShane**
**Reviewer:** @jonathanKingston 
**Asana:** https://app.asana.com/0/1201614831475344/1201880362856602/f

## Description

Added special chars as discussed in the Asana thread

### CODE REVIEW

File: packages/password/index.html [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    <title>Password generator default settings</title>
+</head>
+<body>
+    <h1>Password generator default settings</h1>
+    <pre><code id="rules"></code></pre>
+    <pre><code id="parameters"></code></pre>
+    <pre><code id="password"></code></pre>
+    <script type="module">
+        import {generate} from "./index.js";
+        import {Password} from './lib/apple.password.js'
```
Comment (shakyShane) [inline packages/password/index.html:17]: This is the proof that with ESM, this module can be loaded into a browser environment without any need for a bundler - this makes testing/prototyping/demos much easier

File: packages/password/index.html [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+        const pws = [];
+        for (let i = 0; i < 10; i++) {
+            pws.push(generate())
+        }
+        document.querySelector("#password").textContent = pws.join('\n');
+        document.querySelector("#rules").textContent = Password.defaults.defaultPasswordRules;
+
+        const {parameters, entropy} = new Password().parse(Password.defaults.defaultPasswordRules);
+        document.querySelector("#parameters").textContent = JSON.stringify(parameters, null, 2);
+        document.querySelector("#parameters").textContent += '\n' + "Entropy: " + entropy;
```
Comment (jonathanKingston) [inline packages/password/index.html:27]: Maybe cache this or make it a single line?

File: packages/password/lib/constants.js [hunk 3]
```diff
@@ -1,11 +1,20 @@
 const MIN_LENGTH = 20
 const MAX_LENGTH = 30
-const DEFAULT_PASSWORD_RULES = `minlength: ${MIN_LENGTH}; maxlength: ${MAX_LENGTH};`
+const REQUIRED_CHARS = '-!?$&#%'
```
Comment (jonathanKingston) [inline packages/password/lib/constants.js:3]: Should `MIN_LENGTH`, `MAX_LENGTH` and `REQUIRED_CHARS` all have the `DEFAULT_` prefix?

**Review by jonathanKingston — APPROVED** [review]
SGTM, there's two nits I mentioned feel free to ignore them though.

File: packages/password/lib/constants.js (hunk 3 above)
Comment (shakyShane) [reply packages/password/lib/constants.js]: yep, done

File: packages/password/index.html (hunk 2 above)
Comment (shakyShane) [reply packages/password/index.html]: I split the line out, 👍🏻

File: packages/password/index.html [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+        const passwords = [];
+        for (let i = 0; i < 10; i++) {
+            passwords.push(generate())
+        }
+        elements.password.textContent = passwords.join('\n');
+        elements.rules.textContent = Password.defaults.defaultPasswordRules;
+
+        const {parameters, entropy} = new Password().parse(Password.defaults.defaultPasswordRules);
+        elements.parameters.textContent = JSON.stringify(parameters, null, 2);
+        elements.entropy.textContent += `Entropy: ${entropy}`;
```
Comment (jonathanKingston) [inline packages/password/index.html:34]: This doesn't need to be `+=` anymore.

File: packages/password/tests/generate.test.js [hunk 5]
```diff
@@ -29,6 +30,24 @@ describe('password generation', () => {
...
+        it('creates matches snapshot requirements', () => {
+            const pw = new Password()
+            const { parameters } = pw.parse(constants.DEFAULT_PASSWORD_RULES)
+
+            /**
+             * This snapshot is added as a human-readable check that the internal params
+             * are correct and are not changed by accident.
+             */
+            expect(parameters).toMatchInlineSnapshot(`
+Object {
```
Comment (jonathanKingston) [inline packages/password/tests/generate.test.js:42]: As discussed let's make this JSON :)

**Comment by shakyShane** [comment]
@jonathanKingston sorry for the back and forth, all should be good now

**Review by jonathanKingston — APPROVED** [review]

**Pull request merged**